File: 2026/04-April/23-Thursday/solution.java
// 2615. Sum of Distances

import java.util.*;
 
class Solution {
    public long[] distance(int[] nums) {
        Map<Integer, List<Integer>> map = new HashMap<>();
        
        // Step 1: group indices
        for (int i = 0; i < nums.length; i++) {
            map.computeIfAbsent(nums[i], k -> new ArrayList<>()).add(i);
        }
        
        long[] res = new long[nums.length];
        
        // Step 2: process each group
        for (List<Integer> indices : map.values()) {
            int n = indices.size();
            
            long[] prefix = new long[n + 1];
            for (int i = 0; i < n; i++) {
                prefix[i + 1] = prefix[i] + indices.get(i);
            }
            
            for (int i = 0; i < n; i++) {
                int idx = indices.get(i);
                
                // left side
                long leftCount = i;
                long leftSum = prefix[i];
                long left = idx * leftCount - leftSum;
                
                // right side
                long rightCount = n - i - 1;
                long rightSum = prefix[n] - prefix[i + 1];
                long right = rightSum - idx * rightCount;
                
                res[idx] = left + right;
            }
        }
        
        return res;
    }
}

File: 2026/04-April/23-Thursday/solution.py
# 2615. Sum of Distances

from collections import defaultdict

class Solution:
    def distance(self, nums):
        index_map = defaultdict(list)
        
        # Step 1: group indices by value 
        for i, num in enumerate(nums):
            index_map[num].append(i)
        
        res = [0] * len(nums)
        
        # Step 2: process each group
        for indices in index_map.values():
            n = len(indices)
            
            # prefix sum
            prefix = [0] * (n + 1)
            for i in range(n):
                prefix[i + 1] = prefix[i] + indices[i]
            
            for i in range(n):
                idx = indices[i]
                
                # left side
                left_count = i
                left_sum = prefix[i]
                left = idx * left_count - left_sum
                
                # right side
                right_count = n - i - 1
                right_sum = prefix[n] - prefix[i + 1]
                right = right_sum - idx * right_count
                
                res[idx] = left + right
        
        return res
